arcgis-ios-sdk-samples/Layers/Display KML/DisplayKMLViewController.swift

```diff
@@ -0,0 +1,186 @@
+        case .localFile:
+            changeSourceToLocalFile()
+        case .portalItem:
+            changeSourceToPortalItem()
+        }
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        // Instantiate a map with a dark gray basemap centered on the United States
+        let map = AGSMap(basemapType: .darkGrayCanvasVector, latitude: 39, longitude: -98, levelOfDetail: 4)
+        // Display the map in the map view
+        mapView.map = map
+        
+        // Show the initial KML source
+        updateMapForDisplayedSource()
+        
+        // Add the source code button item to the right of navigation bar.
+        (navigationItem.rightBarButtonItem as! SourceCodeBarButtonItem).filenames = ["DisplayKMLViewController"]
+    }
+    
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        super.prepare(for: segue, sender: sender)
+        
+        // if we're opening the table of data sources
+        if let sourceTableViewController = segue.destination as? DisplayKMLDataSourcesTableViewController {
+            
+            // set the popover height
+            sourceTableViewController.preferredContentSize.height = 135
+            
+            // set the popover delgate
+            sourceTableViewController.popoverPresentationController?.delegate = self
+            
+            // setup the view controller
+            sourceTableViewController.initialSelectedIndex = displayedSource.rawValue
+            sourceTableViewController.selectionHandler = { [weak self] (dataSource) in
+                guard let self = self else{
+                    return
+                }
+                // close the popover
+                self.presentedViewController?.dismiss(animated: true)
+                // set the displayed source based on the selection
+                self.displayedSource = dataSource
+            }
+        }
+    }
+    
+    //MARK: - KML Loading
+    
+    private func display(kmlLayer: AGSKMLLayer) {
+        
+        // Clear the existing layers from the map
+        mapView.map?.operationalLayers.removeAllObjects()
+        // Add the KML layer to the map
+        mapView.map?.operationalLayers.add(kmlLayer)
+        
+        // Show the progress indicator
+        SVProgressHUD.show(withStatus: "Loading KML Layer")
+        
+        // This load call is not required, but it allows for error
+        // feedback and progress indication
+        kmlLayer.load {[weak self] (error) in
+            
+            // Close the progress indicator
+            SVProgressHUD.dismiss()
+
+            if let self = self,
+                let error = error {
+                // Display the error if one occurred
+                let alertController = UIAlertController(title: error.localizedDescription, message: nil, preferredStyle: .alert)
+                alertController.addAction(UIAlertAction(title: "OK", style: .cancel))
```

Comment:
This is a really minor issue, but I think `default` is the appropriate style for an OK button.